File: articles/what-is-decision-science/python/run_all_decision_science_workflows.py
#!/usr/bin/env python3
"""Run all Python workflows for What Is Decision Science?"""

from __future__ import annotations

from pathlib import Path
import subprocess
import sys

ARTICLE_ROOT = Path(__file__).resolve().parents[1]
PYTHON_DIR = ARTICLE_ROOT / "python"

SCRIPTS = [
    "validation_checks.py",
    "expected_value_utility_model.py",
    "decision_tree_diagnostics.py",
    "regret_analysis_minimax.py",
    "robustness_scenario_comparison.py",
    "sensitivity_threshold_analysis.py",
    "decision_record_generator.py",
    "decision_science_diagnostics_workflow.py",
]


def main() -> None:
    for script in SCRIPTS:
        path = PYTHON_DIR / script
        print(f"Running {path.name}...")
        subprocess.run([sys.executable, str(path)], check=True)
    print("All Python decision science workflows completed.")


if __name__ == "__main__":
    main()

File: articles/what-is-decision-science/r/run_all_decision_science_workflows.R
# run_all_decision_science_workflows.R
# Run all R workflows for What Is Decision Science?

args <- commandArgs(trailingOnly = FALSE)
file_arg <- grep("^--file=", args, value = TRUE)

if (length(file_arg) > 0) {
  script_path <- normalizePath(sub("^--file=", "", file_arg[1]), mustWork = TRUE)
  article_root <- normalizePath(file.path(dirname(script_path), ".."), mustWork = TRUE)
} else {
  article_root <- getwd()
}

r_dir <- file.path(article_root, "r")

scripts <- c(
  "decision_science_summary_reporting.R",
  "mcda_tradeoff_profiles.R",
  "sensitivity_weight_diagnostics.R",
  "regret_profile_summary.R",
  "robustness_visualization.R",
  "decision_record_tables.R"
)

for (script in scripts) {
  path <- file.path(r_dir, script)
  message("Running ", path)
  source(path)
}

message("All R decision science workflows completed.")
